yt/frontends/cholla/io.py

```diff
@@ -14,22 +13,24 @@ def _read_particle_coords(self, chunks, ptf):
     def _read_particle_fields(self, chunks, ptf, selector):
         raise NotImplementedError
 
-    def _read_fluid_selection(self, chunks, selector, fields, size):
-        data = {}
-        for field in fields:
-            data[field] = np.empty(size, dtype="float64")
-
-        with h5py.File(self.ds.parameter_filename, "r") as fh:
-            ind = 0
-            for chunk in chunks:
-                for grid in chunk.objs:
-                    nd = 0
-                    for field in fields:
-                        ftype, fname = field
-                        values = fh[fname][:].astype("=f8")
-                        nd = grid.select(selector, values, data[field], ind)
-                    ind += nd
-        return data
+    def io_iter(self, chunks, fields):
+        # this is loosely inspired by the implementation used for Enzo/Enzo-E
+        # - those other options use the lower-level hdf5 interface. Unclear
+        #   whether that affords any advantages...
```

Comment:
Good question.  I think in the past it did because we avoided having to re-allocate temporary scratch space, but I am not sure that would hold up to current inquiries.  I think the big advantage those have is tracking the groups within the iteration.